python/llm/src/ipex_llm/transformers/npu_model.py

```diff
@@ -166,41 +173,66 @@ def from_pretrained(cls, *args, **kwargs):
                 ),
             )
             from ipex_llm.transformers.npu_models.convert_mp import optimize_llm, optimize_llm_pre
-
-            if hasattr(model, "llm"):
-                llm = model.llm
-            else:
-                llm = model
-
-            with torch.no_grad():
-                model.config.update({"mixed_precision": mixed_precision})
-                model.config.update({"group_size": quantization_group_size})
-                optimize_llm_pre(model, qtype, mixed_precision,
-                                 quantization_group_size=quantization_group_size)
-                cls.load_convert(qtype, model, "cpu", modules_to_not_convert,
-                                 quantization_group_size, *args, **kwargs)
-                create_npu_kernels(llm)
-            model = model.eval()
-            logger.info(f"Finish to convert model")
-            model.config.update({"bigdl_transformers_low_bit": qtype})
-            model.share_memory()
-
-            if not pipeline:
-                optimize_llm(
-                    llm,
+            from ipex_llm.transformers.npu_models.convert_mp import optimize_funasr
+            if hasattr(model, "model"):
+                # speech paraformer large
+                encoders = model.model.encoder.encoders[0:31]
+                decoders = model.model.decoder.decoders
+                with torch.no_grad():
+                    cls.load_convert(qtype, encoders,
+                                     "cpu", modules_to_not_convert, *args, **kwargs)
+                    create_npu_kernels(encoders)
+                    cls.load_convert(qtype, decoders,
+                                     "cpu", modules_to_not_convert, *args, **kwargs)
+                    create_npu_kernels(decoders)
+                logger.info(f"Finish to convert model")
+                model.model.share_memory()
+
+                optimize_funasr(
+                    model,
```

Comment:
I think you can refactor these model optimization/conversion logics in to a method (say, `optimize_mode`) in `_BaseAutoModelClass`, which can be implemented in `FunAsrAutoModel` to call `optimize_funasr`. The `from_pretrain` method can just call `optimize_model`